src/reducer/ReducerProducts.jsx

```diff
@@ -0,0 +1,39 @@
+import {useReducer, useEffect} from "react";
+
+import axios from "axios";
+import {reducerFunc} from "../context/ContextProducts";
+
+export const initialState = {
+  loading: false,
+  data: [],
+  error: "",
+};
+
+export function ReducerProducts(getAPI) {
+  const {Api, property} = getAPI;
+
+  const [state, dispatch] = useReducer(
+    reducerFunc,
+    initialState
+  );
+
+  useEffect(() => {
+    (async () => {
+      dispatch({type: "LOADING"});
```

Comment:
This file is not required.

Reply:
I made a reducer component to call API, It needs for that component. Thank you Vishal for PR review